docs/examples/use_cases/tensorflow/efficientdet/train.py

```diff
@@ -0,0 +1,232 @@
+        True,
+        params,
+        strategy if num_devices > 1 else None,
+    )
+
+    if args.eval_after_training or args.eval_during_training:
+        eval_dataset = utils.get_dataset(
+            args,
+            num_devices,
+            False,
+            params,
+            strategy if num_devices > 1 else None,
+        )
+        options = tf.data.Options()
+        options.experimental_distribute.auto_shard_policy = (
+            tf.data.experimental.AutoShardPolicy.DATA
+        )
+        eval_dataset = eval_dataset.with_options(options)
+
+    with strategy.scope():
+        model = efficientdet_net.EfficientDetNet(params=params)
+
+        global_batch_size = args.train_batch_size * strategy.num_replicas_in_sync
+        model.compile(
+            optimizer=optimizers.get_optimizer(
+                params, args.epochs, global_batch_size, args.train_steps
+            )
+        )
+
+        initial_epoch = args.initial_epoch
+        if args.start_weights:
+            image_size = params["image_size"]
+            model.predict(np.zeros((1, image_size[0], image_size[1], 3)))
+            model.load_weights(args.start_weights)
+            fname = args.start_weights.split("/")[-1]
+            ckpt_pattern = f"{args.model_name}\.(\d\d+)\.h5"
+            match = re.match(ckpt_pattern, fname)
+            if match:
+                initial_epoch = int(match.group(1).lstrip("0"))
+
+        callbacks = []
+
+        if args.ckpt_dir:
+            ckpt_dir = args.ckpt_dir
+            if not tf.io.gfile.exists(ckpt_dir):
+                tf.io.gfile.makedirs(tensorboard_dir)
+            callbacks.append(
+                tf.keras.callbacks.ModelCheckpoint(
+                    filepath=os.path.join(
+                        ckpt_dir, "".join([args.model_name, ".{epoch:02d}.h5"])
+                    ),
+                    save_weights_only=True,
+                )
+            )
+
+        if args.log_dir:
+            log_dir = args.log_dir
+            if not tf.io.gfile.exists(log_dir):
+                tf.io.gfile.makedirs(log_dir)
+            callbacks.append(
+                tf.keras.callbacks.TensorBoard(log_dir=log_dir, update_freq="epoch")
+            )
+
+        model.fit(
+            train_dataset,
+            epochs=args.epochs,
+            steps_per_epoch=args.train_steps,
+            initial_epoch=initial_epoch,
+            callbacks=callbacks,
+            validation_data=eval_dataset if args.eval_during_training else None,
+            validation_steps=args.eval_steps,
+            validation_freq=args.eval_freq,
+        )
+
+        if args.eval_after_training:
+            print("Evaluation after training:")
+            model.evaluate(eval_dataset, steps=args.eval_steps)
+
+        model.save_weights(args.output)
+
+
+if __name__ == "__main__":
+    import argparse
+    from argparse_utils import enum_action
+
+    parser = argparse.ArgumentParser()
```

Comment:
General comment to these argument definitions. I believe it would be easier to read, if the descriptions (`help` args) would be unified to start from a capital letter and end with a dot.

Reply:
Fixed.